src/compute/temporal.rs

```diff
@@ -28,6 +28,243 @@ use crate::types::NaturalDataType;
 
 use super::arity::unary;
 
+trait U32Weekday: Datelike {
+    fn u32_weekday(&self) -> u32 {
+        self.weekday().number_from_monday()
+    }
+}
+
+impl U32Weekday for chrono::NaiveDateTime {}
+impl<T: chrono::TimeZone> U32Weekday for chrono::DateTime<T> {}
+
+trait U32IsoWeek: Datelike {
+    fn u32_iso_week(&self) -> u32 {
+        self.iso_week().week()
+    }
+}
+
+impl U32IsoWeek for chrono::NaiveDateTime {}
+impl<T: chrono::TimeZone> U32IsoWeek for chrono::DateTime<T> {}
+
+macro_rules! date_like {
+    ($extract:ident, $array:ident, $data_type:path) => {
+        match $array.data_type() {
```

Comment:
```suggestion
        match $array.data_type().to_logical_type() {
```

It change seems odd, but this is used to "desugar" the `DataType::Extension` type, so that if someone has an extension type built on top of e.g. `Timestamp` type, we also support these methods.

Reply:
Add this method in every place we call `data_type`